test/test_logger_initializer.rb

```diff
@@ -1,46 +1,120 @@
 require_relative 'helper'
 require 'fluent/supervisor'
 require 'fileutils'
+require 'pathname'
 
 class LoggerInitializerTest < ::Test::Unit::TestCase
-  TMP_DIR = File.expand_path(File.dirname(__FILE__) + "/tmp/logger_initializer#{ENV['TEST_ENV_NUMBER']}")
-
-  teardown do
-    begin
-      FileUtils.rm_rf(TMP_DIR)
-    rescue => _
+  def setup
+    @stored_global_logger = $log
+    Dir.mktmpdir do |tmp_dir|
+      begin
+        @tmp_dir = Pathname(tmp_dir)
+        yield
+      ensure
+        $log = @stored_global_logger
```

Comment:
I prefer to place it outside of `Dir.mktmpdir` or in `teardown` to correspond nesting level of replacing `$log`.
(Although probably it's harmless since `Dir.mktmpdir` rarely fail.)

Reply:
Indeed! I will fix it.

Reply:
Fixed! I placed it in `teardown`. Thanks!